File: src/main/java/ch/uzh/ifi/access/model/dto/CourseProgressDTO.java
package ch.uzh.ifi.access.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class CourseProgressDTO {
    private String userId;
    private List<AssignmentProgressDTO> assignments;
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/CourseProgressDTO.kt
package ch.uzh.ifi.access.model.dto

import ch.uzh.ifi.access.projections.CourseInformationPublic


class CourseProgressDTO (
    val userId: String? = null,
    val information: MutableMap<String?, CourseInformationDTO?> = mutableMapOf(),
    val assignments: List<AssignmentProgressDTO>? = null
)

